[게시글 생성/조회] 이지 미션 제출합니다.

skygl opened the pull request:
안녕하세요
첫 미션이다보니 기능 구현을 우선으로 작성했습니다. Annotation에 대한 이해나 테스트에 대해서 고민을 해보면서 Spring을 공부 해야될 것 같습니다.
처음 사용해보는 것이라서 많은 피드백 부탁 드립니다!

andole98 committed:
feat: 미니 미션

andole98 committed:
feat: Article 구현

1. title, contents를 필드로 가짐

//TODO
thymeleaf 템플릿 적용

andole98 committed:
feat: ArticleController 확인

1. thymeleaf 테스트
2. statics resource 테스트

andole98 committed:
feat: Article 조회, 생성 구현

1. thymeleaf 템플릿 th:each 적용
2. ArticleController 변경

andole98 committed:
refactor: HTML 중복 제거

1. common_head
2. common_footer
3. common_js

andole98 committed:
feat: Post, Delete 요청 처리

andole98 committed:
refactor: fragment/common_jun_comment 생성

andole98 committed:
docs: README.md 생성

Commit:
feat: 테스트 코드 작성

woowahan-pjs reviewed (changes requested):
스프링이 익숙지 않음에도 정말 잘 구현해 주셨어요. 👍 
고민해 보면 좋을 것 같은 의견을 남겨 놓았으니 충분히 고민하고 도전해 보세요!
진행 과정에서 궁금한 점은 댓글이나 Slack을 통해 물어보세요.

Commit:
refactor: 자바 컨벤션 수정

- 필요 없는 공백 삭제
- IF문 안에 한 줄이 있어도 중괄호 삽입

Commit:
refactor: Controller 에서 필드 주입 대신 생성자 주입 사용

- ArticleRepository를 필드 주입 받는 코드에서 생성자 주입을 하는 방식 사용
- ArticleControllerTest에서 Redirect시 Header의 Location을 확인하도록 반영

Commit:
refactor: Article 클래스 필드명 수정

- background -> coverUrl

Commit:
refactor: ArticleController 클래스 내 메소드 명 수정

Commit:
refactor: FORM 데이터를 받을 때 ModelAttribute Annotation을 사용하도록 변경

- update, create 메서드에서 RequestParam 대신 ModelAttribute 사용

Commit:
refactor: show, edit 메서드에서 article 객체 데이터를 보내도록 수정

- title, coverUrl, contents를 각각 보내는 방식에서 수정

Commit:
refactor: ArticleDto를 생성하여 Repository에서 Article을 꺼낼 때 사용

- ArticleRepository의 find와 findAll 메서드 실행 시 Dto를 리턴하도록 구현

Commit:
refactor: Article의 필드들을 final로 선언

woowahan-pjs reviewed (changes requested):
피드백을 모두 잘 반영해주셨습니다. 💯 
간단한 생각거리를 남겼으니 코드에 반영해 주셔도 좋습니다.
다음 리뷰 요청 때는 Merge 하도록 하겠습니다.

Commit:
refactor: Article을 Repository에서 가져올 때 DTO 대신 Article을 가져오도록 리팩토링

- ArticleDto 삭제
- findAll() 메서드 실행 시 모든 Article을 갖는 새로운 리스트 리턴
- find() 메서드 실행 시 해당 인덱스의 Article 리턴

woowahan-pjs reviewed (approved):
스프링의 세계에 온 것을 환영합니다. 👍 
이만 Merge 하겠습니다.
